[quant] Fix fusion pattern for add_relu

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39380 [quant][graphmode] Test for another type of ops in insert_observer for if
* #39368 [quant][graphmode][fix] Run preprocess for child module before parent module
* **#39367 [quant] Fix fusion pattern for add_relu**

Summary:
We shouldn't match `%alpha` argument since it could be used by multiple functions

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829295](https://our.internmc.facebook.com/intern/diff/D21829295)

jerryzh168 committed:
[quant] Fix fusion pattern for add_relu

Summary:
We shouldn't match `%alpha` argument since it could be used by multiple functions

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on torch/csrc/jit/passes/quantization/quantization_patterns.h:
```diff
@@ -241,8 +241,7 @@ graph(%a_quant, %packed_params, %r_scale, %r_zero_point, %r_dtype, %stride, %pad
   std::string add_relu = R"(
-graph(%a_quant, %b_quant, %scale, %zero_point, %dtype):
-         %alpha = prim::Constant[value=1]()
+graph(%a_quant, %b_quant, %alpah, %scale, %zero_point, %dtype):
```
should be `alpha`. Do we have test to catch this?

jerryzh168 replied:
yeah, the tests were failing and I was trying to figure out why...

supriyar commented on torch/csrc/jit/passes/quantization/quantization_patterns.h:
```diff
@@ -281,7 +277,7 @@ graph(%a_quant, %b_quant, %scale, %zero_point, %dtype):
 
   std::string quantized_add_relu = R"(
-graph(%a_quant, %b_quant, %scale, %zero_point, %dtype):
+graph(%a_quant, %b_quant, %alpha,  %scale, %zero_point, %dtype):
```
nit, extra space?

supriyar commented:
Should we add or update existing test to catch this in the future?

jerryzh168 committed:
Update on "[quant] Fix fusion pattern for add_relu"


Summary:
We shouldn't match `%alpha` argument since it could be used by multiple functions

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829295](https://our.internmc.facebook.com/intern/diff/D21829295)

[ghstack-poisoned]

jerryzh168 commented:
> Should we add or update existing test to catch this in the future?

yeah we have test for this, I forgot to run the test before submit, please take a look again.

supriyar commented:
> > Should we add or update existing test to catch this in the future?
> 
> yeah we have test for this, I forgot to run the test before submit, please take a look again.

I was referring to testing the case where `alpha` is used in multiple places.

jerryzh168 commented:
> I was referring to testing the case where `alpha` is used in multiple places.

Oh, I see, yeah I can add a test for this

jerryzh168 committed:
Update on "[quant] Fix fusion pattern for add_relu"


Summary:
We shouldn't match `%alpha` argument since it could be used by multiple functions

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829295](https://our.internmc.facebook.com/intern/diff/D21829295)

[ghstack-poisoned]
